src/backend/joanie/tests/core/api/order/test_retract.py

```diff
@@ -0,0 +1,138 @@
+"""Tests for the Order withdraw API."""
+
+from datetime import datetime
+from http import HTTPStatus
+from unittest import mock
+
+from django.core.cache import cache
+
+from joanie.core import enums, factories
+from joanie.tests.base import BaseAPITestCase
+
+
+class OrderWithdrawApiTest(BaseAPITestCase):
+    """Test the API of the Order withdraw endpoint."""
+
+    maxDiff = None
+
+    def setUp(self):
+        """Clear cache after each tests"""
+        cache.clear()
+
+    def test_api_order_withdraw_anonymous(self):
+        """
+        Anonymous user cannot withdraw order
+        """
+        order = factories.OrderFactory()
+
+        response = self.client.post(
+            f"/api/v1.0/orders/{order.id}/withdraw/",
+            content_type="application/json",
+        )
+
+        self.assertEqual(response.status_code, HTTPStatus.UNAUTHORIZED)
+        order.refresh_from_db()
+        self.assertNotEqual(order.state, enums.ORDER_STATE_CANCELED)
+
+    def test_api_order_withdraw_authenticated_unexisting(self):
+        """
+        User should receive 404 when withdrawing a non existing order
+        """
+        user = factories.UserFactory()
+        token = self.generate_token_from_user(user)
+
+        response = self.client.post(
+            "/api/v1.0/orders/notarealid/withdraw/",
```

Comment:
You should use a random `uuid`. I think here the 404 is juste because it does not match a route. Not because the object does not exists in database.